app/assets/javascripts/account/directives/myApplicationDirective.js.coffee

```diff
@@ -54,11 +60,51 @@ angular.module('dahlia.directives')
       scope.isSubmitted = ->
         ShortFormApplicationService.applicationWasSubmitted(scope.application)
 
+      scope.submittedWithLotteryResults = ->
+        scope.isSubmitted() && scope.listing.Lottery_Results
+
       scope.isPastDue = ->
         moment(scope.listing.Application_Due_Date) < moment()
 
       scope.lotteryNumber = ->
-        { lotteryNumber: scope.application.lotteryNumber }
+        if scope.listing.Lottery_Results
+          html = """
+            <button class='button-link lined' ng-click='viewLotteryResults()'>
+              ##{scope.application.lotteryNumber}
+            </button>
+          """
+          $sce.trustAsHtml(html)
+        else
+          "##{scope.application.lotteryNumber}"
+
+      scope.viewLotteryResults = ->
+        # if the search failed, then viewLotteryResults becomes a button to open the PDF instead
+        if ListingService.error.lotteryRank && scope.listing.LotteryResultsURL
+          $window.open(scope.listing.LotteryResultsURL, '_blank')
+          return
+
+        ShortFormNavigationService.isLoading(true)
+        # have to setup our "current" listing and application in the Services for the modal to play nicely
+        ListingService.loadListing(scope.listing)
+        angular.copy(scope.application, ShortFormApplicationService.application)
+        # lookup individual lottery ranking and then open the modal
+        ListingService.getLotteryRanking(scope.application.lotteryNumber).then(->
+          ListingService.openLotteryResultsModal()
+          ShortFormNavigationService.isLoading(false)
+        ).catch(->
+          ###
+          # NOTE: Even though we have the listing already "loaded" via API AccountController `map_listings_to_applications`
+          # this has one limitation which is that by using ListingService.listings() ("browse" API) we do not get LotteryResultsURL.
+          # So in the event that getLotteryRanking fails and we want to fall back to LotteryResultsURL, we have to call getListing.
+          ###
+          ListingService.loadListing({})
+          # have to restart the loader because the error would have stopped it
+          ShortFormNavigationService.isLoading(true)
+          ListingService.getListing(scope.listing.Id).then ->
+            # alert message: "Oops! Please download lottery results instead..." ??
```

Comment:
what does this comment pertain to?

Reply:
This is Dave's work. My best guess is that it references the uncertainty at the time about what kinds of error states or fallback options there should be in case of failure to load certain objects in this process. If you read the comments on the Pivotal story (https://www.pivotaltracker.com/story/show/152122952) you can get a sense of the discussion around this. I don't feel strongly either way about leaving this comment in or removing it. It may be better to remove it to avoid confusion. The error states issue is captured in a Pivotal story (https://www.pivotaltracker.com/story/show/152730034), so I don't think that we would lose any reminder about this by removing the comment.

Reply:
I would like to remove it. Since you're out of hours, I'll do this and merge. Thanks!